substrate/frame/atomic-swap/src/lib.rs

```diff
@@ -54,22 +54,10 @@ use frame::{
 	prelude::*,
 	traits::{BalanceStatus, Currency, ReservableCurrency},
 };
-use scale_info::TypeInfo;
 
 /// Pending atomic swap operation.
-#[derive(
-	Clone,
-	Eq,
-	PartialEq,
-	RuntimeDebugNoBound,
-	Encode,
-	Decode,
-	DecodeWithMemTracking,
-	TypeInfo,
-	MaxEncodedLen,
-)]
-#[scale_info(skip_type_params(T))]
-#[codec(mel_bound())]
+#[stored(skip(T))]
+#[derive(DecodeWithMemTracking)]
```

Comment:
Personally I would include this into the default, potentially some type don't implement it but we can just implement it by default IMO.